src/api/app/jobs/report_to_scm_job.rb

```diff
@@ -0,0 +1,32 @@
+class ReportToScmJob < CreateJob
+  ALLOWED_EVENTS = ['Event::BuildFail', 'Event::BuildSuccess'].freeze
+
+  # We don't properly capitalize SCM in the class name since CreateJob is doing `CLASS_NAME.to_s.camelize.safe_constantize`
+  queue_as :scm
+
+  def perform(event_id)
+    event = Event::Base.find(event_id)
+    return false unless event
+
+    return false unless event.undone_jobs.positive?
+
+    event_type = event.eventtype
+    return false unless ALLOWED_EVENTS.include?(event_type)
+
+    event_package = Package.find_by_project_and_name(event.payload['project'], event.payload['package'])
+    return false if event_package.blank?
+
+    EventSubscriptionsFinder.new
+                            .for_scm_channel_with_token(event_type: event_type, event_package: event_package)
+                            .each do |event_subscription|
+      SCMStatusReporter.new(event_subscription.payload,
+                            event_subscription.token.scm_token,
+                            event_subscription.eventtype).call
+    rescue StandardError => e
+      Airbrake.notify(e, event_id: event.id)
+      event.mark_job_done!
```

Comment:
Why mark it was done in case of error?

Reply:
If somehow the job crashes due to an error, the `event.undone_jobs` field will never be changed in the `after_perform` callback from the parent class `CreateJob` (see code snippet below). A `ReportToScmJob` job would then run again for the event and keep on crashing.

https://github.com/openSUSE/open-build-service/blob/1379e7286ef419c56bc4f6f71a95bbdda79d2400/src/api/app/jobs/create_job.rb#L6-L15

Reply:
Yes, and maybe it would run again an not crash?